src/vsr/replica.zig

```diff
@@ -4360,18 +4361,25 @@ pub fn ReplicaType(
                     .op = entry.header.op,
                     .context = entry.header.checksum,
                 });
-                // Don't return here — it is safe to start a view without all replies repaired.
             }
 
-            // Commit ops, which may in turn discover faulty prepares and drive more repairs:
             if (self.commit_min < self.commit_max) {
+                // Try to the commit prepares we already have, even if we don't have all of them.
+                // This helps when a replica is recovering from a crash and has a mostly intact
+                // journal, with just some prepares missing. We do have the headers and know
+                // that they form a valid hashcahin. Committing may discover more faulty prepares
+                // and drive further repairs.
                 assert(!self.solo());
                 self.commit_journal(self.commit_max);
-                return;
             }
-            assert(self.commit_max <= self.op);
 
-            if (self.status == .view_change and self.primary_index(self.view) == self.replica) {
+            if (self.status == .view_change and
+                self.primary_index(self.view) == self.replica and
+                self.commit_min == self.commit_max)
+            {
+                assert(self.journal.dirty.count == 0);
```

Comment:
I was a little concerned that this assert could be hit, because `repair_prepares` cleans up the "final" dirty bits [at the end](https://github.com/tigerbeetledb/tigerbeetle/blob/4e724158e006d7eb70b362aae84f28a9676527a6/src/vsr/replica.zig#L4809-L4838).

But I think its ok since `repair_prepares` would only early-return if all of the journal's write budget is used, and those writes imply `dirty > 0`.